bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/addons/AddonResource.java

```diff
@@ -389,8 +391,13 @@ private void postFailureEvent(String addonId, @Nullable String msg) {
     }
 
     private @Nullable AddonService getDefaultService() {
-        return addonServices.stream().filter(addonService -> DEFAULT_ADDON_SERVICE.equals(addonService.getId()))
-                .findFirst().orElse(addonServices.stream().findFirst().orElse(null));
+        Optional<AddonService> karafAddonService = addonServices.stream()
+                .filter(addonService -> KARAF_ADDON_SERVICE_ID.equals(addonService.getId())).findFirst();
+        Optional<AddonService> eclipseAddonService = addonServices.stream()
+                .filter(addonService -> ECLIPSE_ADDON_SERVICE_ID.equals(addonService.getId())).findFirst();
+        Optional<AddonService> firstAddonService = addonServices.stream().findFirst();
+        return Stream.of(karafAddonService, eclipseAddonService, firstAddonService).filter(Optional::isPresent)
+                .findFirst().get().orElse(null);
```

Comment:
Why is this needed? If the eclipse add-on service prefixes the ids with it's own id (like the other add-on services do) it should work without this strong coupling.

Reply:
Yes that was something already on my todo list and it seems to work.

Though it could probably use some UI tweaks as well:

![image](https://user-images.githubusercontent.com/12213581/221406139-8e250e8b-d5cc-4d3b-adff-f7c7f994e9a8.png)